setup.js

```diff
@@ -0,0 +1,184 @@
+                    console.log('\nWriting files...');
+
+                    // Remove prettier stuff
+                    try {
+                      fs.unlinkSync('package.json');
+                      fs.unlinkSync('package-lock.json');
+                      fs.rmSync('node_modules', { recursive: true });
+                    } catch (error) {
+                      if (error.code !== 'ENOENT' && error.code !== 'EEXIST') {
+                        console.error(error);
+                        process.exit(1);
+                      }
+                    }
+
+                    // Writing general stuff
+                    const filesToUpdate = [
+                      'LICENSE',
+                      'CITATION.cff',
+                    ];
+                    filesToUpdate.forEach((fileName) => {
+                      try {
+                        let fileContent = fs.readFileSync(`./template/${fileName}`, 'utf8');
+                        fileContent = fileContent.replace(/{{REPOSITORY}}/g, `${username}/${repository}`)
+                          .replace(/{{PROJECT_NAME}}/g, proj_name)
+                          .replace(/{{PROJECT_SHORT_DESCRIPTION}}/g, proj_short_desc)
+                          .replace(/{{PROJECT_LONG_DESCRIPTION}}/g, proj_long_desc)
+                          .replace(/{{DOCS_URL}}/g, docs_url)
+                          .replace(/{{EMAIL}}/g, email)
+                          .replace(/{{USERNAME}}/g, username)
+                          .replace(/{{NAME}}/g, name);
+                        fs.writeFileSync(`./template/${fileName}`, fileContent);
+                      } catch (error) {
+                        if (error.code !== 'ENOENT' && error.code !== 'EEXIST') {
+                          console.error(error);
+                          process.exit(1);
+                        } else {
+                            console.log(`File ${fileName} not found.`);
+                        }
+                      }
+                    });
+
+                    // Write CODEOWNERS
+                    try {
+                      fs.appendFileSync('./template/.github/CODEOWNERS', `* @${username}`);
+                    } catch (error) {
+                      if (error.code !== 'ENOENT' && error.code !== 'EEXIST') {
+                        console.error(error);
+                        process.exit(1);
+                      } else {
+                        fs.renameSync('./template/.github/CODEOWNERS', '.github/CODEOWNERS');
+                      }
+                    }
+
+                    // Optional keep up-to-date
+                    rl.question('Would you like to keep up-to-date with the template? (y/n)\n=> ', (up_to_date) => {
+                      if (up_to_date.toLowerCase() === 'y') {
+                        console.log('Writing ignore file...');
+                        try {
+                          fs.appendFileSync('./template/.templatesyncignore', `
```

Comment:
This would do better as a variable at the top of the file under a "template section" or equivalent.

Something like this
```js
const TEMPLATE_SYNC = `
...
...
...`

// many lines of code

fs.appendFileSync('...', TEMPLATE_SYNC)
```